### ISSUE TITLE
Installation issue: pasta using wrong Python executable

### ISSUE THREAD

**Issue by ghost** [LLNL/spack#32485]
### Steps to reproduce the issue

```console
$ spack install pasta
```

### Error message

<details><summary>Error message</summary><pre>
==> [2022-09-01-16:32:52.547994] Installing . to /modules/spack/opt/spack/linux-ubuntu20.04-cascadelake/gcc-9.4.0/pasta-1.8.3-wi4vc2vu6yg4uvu6hmpfhkdvo4kej>==> [2022-09-01-16:32:59.364186] '/usr/bin/python2.7' 'setup.py' 'develop'
ImportError: No module named site
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/python2.7' 'setup.py' 'develop'
See build log for details:
  /scratch/astroth-cpu004/3304620/mgjanci_umass_edu/spack-stage/spack-stage-pasta-1.8.3-wi4vc2vu6yg4uvu6hmpfhkdvo4kejwdp/spack-build-out.txt
</pre></details>

### Information on your system

```
* **Spack:** 0.18.0 (c09bf37ff690c29779a342670cf8a171ad1b9233)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-ivybridge
* **Concretizer:** clingo
```

### Additional information

```
BASH_ENV=/modules/lmod/lmod/lmod/init/bash; export BASH_ENV
CC=/modules/spack/lib/spack/env/gcc/gcc; export CC
CLASSPATH=''; export CLASSPATH
CMAKE_PREFIX_PATH=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/py-dendropy-4.3.0-c46wd4rqalc6tfn62wav5zqm3e4vzw2p:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/python-3.8.12-4cdo3s4ubyx5uct6w6x2a7nipxkjowvx:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/util-linux-uuid-2.36.2-yviacyriovepljn5yspyr6qmxjrgvbvl:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/sqlite-3.36.0-shfz5o6exs27abcdtvzwexzl3wm2abnu:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openssl-1.1.1l-tgeombqxtfao2uanglbjfxjuo4xh2voh:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libffi-3.3-3j4eg6mt6c4s4vtv7lgv57jvokm3igdj:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gettext-0.21-st467yhuuvdbf5s7jemnjcngud2geeow:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/tar-1.34-e4wjcd5tsosgaxo6gtxv2gaqu2qhp5dv:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libxml2-2.9.12-comayyuhynetd6n7gezhibyvpxkstwee:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/zlib-1.2.11-bt5eklpsblqwhnh6jnz3u7gknp4dkodx:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/xz-5.2.5-coej4nkhb3rthncxmidlj4zexh55s4tg:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libiconv-1.16-mohb7zcc55othywubbgabbfqzpwkkcvx:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gdbm-1.19-gyq7i3hir3m6ooly5uczfyunzqim66ih:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/readline-8.1-ti4njcb6thpppwjr43kvxtzdx33rs47w:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/ncurses-6.2-24quon5dqd6nio5qlsuuorsjjdffbfc7:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/expat-2.4.1-yjpjginbmljftnhwatiexwaucj4z5d3e:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libbsd-0.11.3-vxsqzljtd3zugm2bo6khw2orvd7ttm3z:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libmd-1.0.3-p4upbcopgy3ebdqpcsxnnnalx5wrjzss:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/bzip2-1.0.8-wzfp635na43r2urothaasdtzezh2jsxt:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openjdk-11.0.12_7-bw3ad6wmn46klm3z7rourm775xe3ce7t; export CMAKE_PREFIX_PATH
CXX=/modules/spack/lib/spack/env/gcc/g++; export CXX
ENVIRONMENT=BATCH; export ENVIRONMENT
F77=/modules/spack/lib/spack/env/gcc/gfortran; export F77
FC=/modules/spack/lib/spack/env/gcc/gfortran; export FC
HOME=/home/mgjanci_umass_edu; export HOME
HOSTNAME=login-build001; export HOSTNAME
JAVA_HOME=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openjdk-11.0.12_7-bw3ad6wmn46klm3z7rourm775xe3ce7t; export JAVA_HOME
LANG=C.UTF-8; export LANG
LC_ALL=C; export LC_ALL
LDCXXSHARED='/modules/spack/lib/spack/env/gcc/g++ -shared'; export LDCXXSHARED
LDSHARED='/modules/spack/lib/spack/env/gcc/gcc -shared -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/bzip2-1.0.8-wzfp635na43r2urothaasdtzezh2jsxt/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/expat-2.4.1-yjpjginbmljftnhwatiexwaucj4z5d3e/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gdbm-1.19-gyq7i3hir3m6ooly5uczfyunzqim66ih/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gettext-0.21-st467yhuuvdbf5s7jemnjcngud2geeow/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libffi-3.3-3j4eg6mt6c4s4vtv7lgv57jvokm3igdj/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/ncurses-6.2-24quon5dqd6nio5qlsuuorsjjdffbfc7/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openssl-1.1.1l-tgeombqxtfao2uanglbjfxjuo4xh2voh/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/readline-8.1-ti4njcb6thpppwjr43kvxtzdx33rs47w/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/sqlite-3.36.0-shfz5o6exs27abcdtvzwexzl3wm2abnu/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/util-linux-uuid-2.36.2-yviacyriovepljn5yspyr6qmxjrgvbvl/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/xz-5.2.5-coej4nkhb3rthncxmidlj4zexh55s4tg/lib -L/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/zlib-1.2.11-bt5eklpsblqwhnh6jnz3u7gknp4dkodx/lib'; export LDSHARED
LMOD_CMD=/modules/lmod/lmod/lmod/libexec/lmod; export LMOD_CMD
LMOD_DIR=/modules/lmod/lmod/lmod/libexec; export LMOD_DIR
LMOD_PKG=/modules/lmod/lmod/lmod; export LMOD_PKG
LMOD_SETTARG_FULL_SUPPORT=no; export LMOD_SETTARG_FULL_SUPPORT
LMOD_VERSION=8.6.17; export LMOD_VERSION
LMOD_sys=Linux; export LMOD_sys
LOGNAME=mgjanci_umass_edu; export LOGNAME
MAIL=/var/mail/mgjanci_umass_edu; export MAIL
MANPATH=/modules/lmod/lmod/lmod/share/man:; export MANPATH
MODULEPATH=/modules/spack/share/spack/modules/linux-ubuntu20.04-x86_64:/modules/modulefiles; export MODULEPATH
MODULESHOME=/modules/lmod/lmod/lmod; export MODULESHOME
PASTA_TOOLS_DEVDIR=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/pasta-1.8.3-2ohdzqsdrfyusyjtxaxe6janiow5ziwx/sate-tools-linux; export PASTA_TOOLS_DEVDIR
PATH=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/py-dendropy-4.3.0-c46wd4rqalc6tfn62wav5zqm3e4vzw2p/bin:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/python-3.8.12-4cdo3s4ubyx5uct6w6x2a7nipxkjowvx/bin:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/python-3.8.12-4cdo3s4ubyx5uct6w6x2a7nipxkjowvx/bin:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openjdk-11.0.12_7-bw3ad6wmn46klm3z7rourm775xe3ce7t/bin:/modules/spack/lib/spack/env/gcc:/modules/spack/lib/spack/env/case-insensitive:/modules/spack/lib/spack/env:/modules/spack/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin; export PATH
PKG_CONFIG_PATH=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/python-3.8.12-4cdo3s4ubyx5uct6w6x2a7nipxkjowvx/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/util-linux-uuid-2.36.2-yviacyriovepljn5yspyr6qmxjrgvbvl/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/sqlite-3.36.0-shfz5o6exs27abcdtvzwexzl3wm2abnu/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openssl-1.1.1l-tgeombqxtfao2uanglbjfxjuo4xh2voh/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libffi-3.3-3j4eg6mt6c4s4vtv7lgv57jvokm3igdj/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libxml2-2.9.12-comayyuhynetd6n7gezhibyvpxkstwee/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/zlib-1.2.11-bt5eklpsblqwhnh6jnz3u7gknp4dkodx/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/xz-5.2.5-coej4nkhb3rthncxmidlj4zexh55s4tg/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/readline-8.1-ti4njcb6thpppwjr43kvxtzdx33rs47w/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/ncurses-6.2-24quon5dqd6nio5qlsuuorsjjdffbfc7/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/expat-2.4.1-yjpjginbmljftnhwatiexwaucj4z5d3e/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libbsd-0.11.3-vxsqzljtd3zugm2bo6khw2orvd7ttm3z/lib/pkgconfig:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libmd-1.0.3-p4upbcopgy3ebdqpcsxnnnalx5wrjzss/lib/pkgconfig; export PKG_CONFIG_PATH
#PWD=/modules/spack-package-installers; export PWD
PYTHONHOME=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/python-3.8.12-4cdo3s4ubyx5uct6w6x2a7nipxkjowvx; export PYTHONHOME
PYTHONPATH=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/py-dendropy-4.3.0-c46wd4rqalc6tfn62wav5zqm3e4vzw2p/lib/python3.8/site-packages; export PYTHONPATH
SHELL=/bin/bash; export SHELL
SHLVL=2; export SHLVL
SLURMD_NODENAME=login-build001; export SLURMD_NODENAME
SLURM_CLUSTER_NAME=unity; export SLURM_CLUSTER_NAME
SLURM_CONF=/var/spool/slurm/slurmd/conf-cache/slurm.conf; export SLURM_CONF
SLURM_CPUS_ON_NODE=8; export SLURM_CPUS_ON_NODE
SLURM_CPUS_PER_TASK=8; export SLURM_CPUS_PER_TASK
SLURM_EXPORT_ENV=SPACK_INSTALL_ARGS=pasta; export SLURM_EXPORT_ENV
SLURM_GET_USER_ENV=1; export SLURM_GET_USER_ENV
SLURM_GTIDS=0; export SLURM_GTIDS
SLURM_JOBID=3304985; export SLURM_JOBID
SLURM_JOB_ACCOUNT=pi_jgriffin_umass_edu; export SLURM_JOB_ACCOUNT
SLURM_JOB_CPUS_PER_NODE=8; export SLURM_JOB_CPUS_PER_NODE
SLURM_JOB_GID=1643; export SLURM_JOB_GID
SLURM_JOB_ID=3304985; export SLURM_JOB_ID
SLURM_JOB_NAME=build_pasta; export SLURM_JOB_NAME
SLURM_JOB_NODELIST=login-build001; export SLURM_JOB_NODELIST
SLURM_JOB_NUM_NODES=1; export SLURM_JOB_NUM_NODES
SLURM_JOB_PARTITION=building; export SLURM_JOB_PARTITION
SLURM_JOB_QOS=normal; export SLURM_JOB_QOS
SLURM_JOB_UID=1643; export SLURM_JOB_UID
SLURM_JOB_USER=mgjanci_umass_edu; export SLURM_JOB_USER
SLURM_LOCALID=0; export SLURM_LOCALID
SLURM_MEM_PER_CPU=1024; export SLURM_MEM_PER_CPU
SLURM_NNODES=1; export SLURM_NNODES
SLURM_NODEID=0; export SLURM_NODEID
SLURM_NODELIST=login-build001; export SLURM_NODELIST
SLURM_NODE_ALIASES='(null)'; export SLURM_NODE_ALIASES
SLURM_PRIO_PROCESS=0; export SLURM_PRIO_PROCESS
SLURM_PROCID=0; export SLURM_PROCID
SLURM_SUBMIT_DIR=/modules/spack-package-installers; export SLURM_SUBMIT_DIR
SLURM_SUBMIT_HOST=login2; export SLURM_SUBMIT_HOST
SLURM_TASKS_PER_NODE=1; export SLURM_TASKS_PER_NODE
SLURM_TASK_PID=2405361; export SLURM_TASK_PID
SLURM_TOPOLOGY_ADDR=login-build001; export SLURM_TOPOLOGY_ADDR
SLURM_TOPOLOGY_ADDR_PATTERN=node; export SLURM_TOPOLOGY_ADDR_PATTERN
SLURM_WORKING_CLUSTER=unity:slurmctld1:6817:8960:109; export SLURM_WORKING_CLUSTER
SPACK_CC=/usr/bin/gcc; export SPACK_CC
SPACK_CC_RPATH_ARG=-Wl,-rpath,; export SPACK_CC_RPATH_ARG
SPACK_COMPILER_SPEC=gcc@9.4.0; export SPACK_COMPILER_SPEC
SPACK_CUR_ARCH=linux-ubuntu20.04-x86_64; export SPACK_CUR_ARCH
SPACK_CXX=/usr/bin/g++; export SPACK_CXX
SPACK_CXX_RPATH_ARG=-Wl,-rpath,; export SPACK_CXX_RPATH_ARG
SPACK_DEBUG_LOG_DIR=/modules/spack-package-installers; export SPACK_DEBUG_LOG_DIR
SPACK_DEBUG_LOG_ID=pasta-2ohdzqs; export SPACK_DEBUG_LOG_ID
SPACK_DTAGS_TO_ADD=--disable-new-dtags; export SPACK_DTAGS_TO_ADD
SPACK_DTAGS_TO_STRIP=--enable-new-dtags; export SPACK_DTAGS_TO_STRIP
SPACK_ENV_PATH=/modules/spack/lib/spack/env:/modules/spack/lib/spack/env/case-insensitive:/modules/spack/lib/spack/env/gcc; export SPACK_ENV_PATH
SPACK_F77=/usr/bin/gfortran; export SPACK_F77
SPACK_F77_RPATH_ARG=-Wl,-rpath,; export SPACK_F77_RPATH_ARG
SPACK_FC=/usr/bin/gfortran; export SPACK_FC
SPACK_FC_RPATH_ARG=-Wl,-rpath,; export SPACK_FC_RPATH_ARG
SPACK_INCLUDE_DIRS=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/python-3.8.12-4cdo3s4ubyx5uct6w6x2a7nipxkjowvx/include/python3.8:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/util-linux-uuid-2.36.2-yviacyriovepljn5yspyr6qmxjrgvbvl/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/sqlite-3.36.0-shfz5o6exs27abcdtvzwexzl3wm2abnu/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openssl-1.1.1l-tgeombqxtfao2uanglbjfxjuo4xh2voh/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libffi-3.3-3j4eg6mt6c4s4vtv7lgv57jvokm3igdj/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gettext-0.21-st467yhuuvdbf5s7jemnjcngud2geeow/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libxml2-2.9.12-comayyuhynetd6n7gezhibyvpxkstwee/include/libxml2:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libxml2-2.9.12-comayyuhynetd6n7gezhibyvpxkstwee/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/zlib-1.2.11-bt5eklpsblqwhnh6jnz3u7gknp4dkodx/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/xz-5.2.5-coej4nkhb3rthncxmidlj4zexh55s4tg/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libiconv-1.16-mohb7zcc55othywubbgabbfqzpwkkcvx/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gdbm-1.19-gyq7i3hir3m6ooly5uczfyunzqim66ih/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/readline-8.1-ti4njcb6thpppwjr43kvxtzdx33rs47w/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/ncurses-6.2-24quon5dqd6nio5qlsuuorsjjdffbfc7/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/expat-2.4.1-yjpjginbmljftnhwatiexwaucj4z5d3e/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libbsd-0.11.3-vxsqzljtd3zugm2bo6khw2orvd7ttm3z/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libmd-1.0.3-p4upbcopgy3ebdqpcsxnnnalx5wrjzss/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/bzip2-1.0.8-wzfp635na43r2urothaasdtzezh2jsxt/include:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openjdk-11.0.12_7-bw3ad6wmn46klm3z7rourm775xe3ce7t/include; export SPACK_INCLUDE_DIRS
SPACK_INSTALL_ARGS=pasta; export SPACK_INSTALL_ARGS
SPACK_LINKER_ARG=-Wl,; export SPACK_LINKER_ARG
SPACK_LINK_DIRS=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/py-dendropy-4.3.0-c46wd4rqalc6tfn62wav5zqm3e4vzw2p/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/python-3.8.12-4cdo3s4ubyx5uct6w6x2a7nipxkjowvx/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/util-linux-uuid-2.36.2-yviacyriovepljn5yspyr6qmxjrgvbvl/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/sqlite-3.36.0-shfz5o6exs27abcdtvzwexzl3wm2abnu/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openssl-1.1.1l-tgeombqxtfao2uanglbjfxjuo4xh2voh/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libffi-3.3-3j4eg6mt6c4s4vtv7lgv57jvokm3igdj/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gettext-0.21-st467yhuuvdbf5s7jemnjcngud2geeow/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libxml2-2.9.12-comayyuhynetd6n7gezhibyvpxkstwee/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/zlib-1.2.11-bt5eklpsblqwhnh6jnz3u7gknp4dkodx/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/xz-5.2.5-coej4nkhb3rthncxmidlj4zexh55s4tg/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libiconv-1.16-mohb7zcc55othywubbgabbfqzpwkkcvx/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gdbm-1.19-gyq7i3hir3m6ooly5uczfyunzqim66ih/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/readline-8.1-ti4njcb6thpppwjr43kvxtzdx33rs47w/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/ncurses-6.2-24quon5dqd6nio5qlsuuorsjjdffbfc7/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/expat-2.4.1-yjpjginbmljftnhwatiexwaucj4z5d3e/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libbsd-0.11.3-vxsqzljtd3zugm2bo6khw2orvd7ttm3z/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libmd-1.0.3-p4upbcopgy3ebdqpcsxnnnalx5wrjzss/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/bzip2-1.0.8-wzfp635na43r2urothaasdtzezh2jsxt/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openjdk-11.0.12_7-bw3ad6wmn46klm3z7rourm775xe3ce7t/lib/server:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openjdk-11.0.12_7-bw3ad6wmn46klm3z7rourm775xe3ce7t/lib; export SPACK_LINK_DIRS
SPACK_PYTHON=/usr/bin/python3; export SPACK_PYTHON
SPACK_ROOT=/modules/spack; export SPACK_ROOT
SPACK_RPATH_DIRS=/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/pasta-1.8.3-2ohdzqsdrfyusyjtxaxe6janiow5ziwx/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/pasta-1.8.3-2ohdzqsdrfyusyjtxaxe6janiow5ziwx/lib64:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/py-dendropy-4.3.0-c46wd4rqalc6tfn62wav5zqm3e4vzw2p/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/python-3.8.12-4cdo3s4ubyx5uct6w6x2a7nipxkjowvx/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/util-linux-uuid-2.36.2-yviacyriovepljn5yspyr6qmxjrgvbvl/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/sqlite-3.36.0-shfz5o6exs27abcdtvzwexzl3wm2abnu/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openssl-1.1.1l-tgeombqxtfao2uanglbjfxjuo4xh2voh/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libffi-3.3-3j4eg6mt6c4s4vtv7lgv57jvokm3igdj/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gettext-0.21-st467yhuuvdbf5s7jemnjcngud2geeow/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libxml2-2.9.12-comayyuhynetd6n7gezhibyvpxkstwee/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/zlib-1.2.11-bt5eklpsblqwhnh6jnz3u7gknp4dkodx/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/xz-5.2.5-coej4nkhb3rthncxmidlj4zexh55s4tg/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libiconv-1.16-mohb7zcc55othywubbgabbfqzpwkkcvx/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/gdbm-1.19-gyq7i3hir3m6ooly5uczfyunzqim66ih/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/readline-8.1-ti4njcb6thpppwjr43kvxtzdx33rs47w/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/ncurses-6.2-24quon5dqd6nio5qlsuuorsjjdffbfc7/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/expat-2.4.1-yjpjginbmljftnhwatiexwaucj4z5d3e/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libbsd-0.11.3-vxsqzljtd3zugm2bo6khw2orvd7ttm3z/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/libmd-1.0.3-p4upbcopgy3ebdqpcsxnnnalx5wrjzss/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/bzip2-1.0.8-wzfp635na43r2urothaasdtzezh2jsxt/lib:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openjdk-11.0.12_7-bw3ad6wmn46klm3z7rourm775xe3ce7t/lib/server:/modules/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.4.0/openjdk-11.0.12_7-bw3ad6wmn46klm3z7rourm775xe3ce7t/lib; export SPACK_RPATH_DIRS
SPACK_SHORT_SPEC='pasta@1.8.3%gcc@9.4.0 arch=linux-ubuntu20.04-x86_64/2ohdzqs'; export SPACK_SHORT_SPEC
SPACK_SYSTEM_DIRS=/bin:/usr/bin:/usr/local/bin:/bin64:/usr/bin64:/usr/local/bin64:/include:/usr/include:/usr/local/include:/lib:/usr/lib:/usr/local/lib:/lib64:/usr/lib64:/usr/local/lib64:/:/usr:/usr/local; export SPACK_SYSTEM_DIRS
SPACK_TARGET_ARGS='-march=x86-64 -mtune=generic'; export SPACK_TARGET_ARGS
TEMP=/scratch/login-build001/3304985; export TEMP
TEMPDIR=/scratch/login-build001/3304985; export TEMPDIR
TMP=/scratch/login-build001/3304985; export TMP
TMPDIR=/scratch/login-build001/3304985; export TMPDIR
USER=mgjanci_umass_edu; export USER
XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop; export XDG_DATA_DIRS
_=/modules/spack/bin/spack; export _
```

Our system by default has python 3.8 installed. But we also have 2.7.18 installed as a module through spack. When installing PASTA through spack we receive an error msg about site.py which is a python 2.7 import. This happens because the PYTHONHOME path is set to our python 3.8 directory even though in the build for PASTA it requires python 2.7 as a build dependency and calls which(python2.7).  Above I have attached the build env.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
What does `spack spec pasta` show? It seems like Python 3 is being used, not Python 2, so `PYTHONHOME` is set correctly. According to the build recipe, it seems like pasta 1.9.0 supports Python 3 only. If the build script is still using Python 2, maybe there's a bug in the build script?

@snehring

**Comment by snehring** [comment]
This is likely fixed by #32077 and then fixed better by #32220. Prior to that the recipe called which("python2.7") which was an artifact of an earlier version.

**Comment by ghost** [comment]
We were using the un-updated package file that was not coded to install pasta 1.9 with python 3 but when I imported the updated package file as a custom and ran it with spack it worked as intended. Thank you for the help.
